File: leetsync/3753-maximum-difference-between-even-and-odd-frequency-i/maximum-difference-between-even-and-odd-frequency-i.go
import "math"

func maxDifference(s string) int {
    freq := make([]int, 26)
    a := 'a'

    for _, ch := range s {
        freq[ch-a]++
    }

    odd := 0
    even := math.MaxInt32

    for _, n := range freq {
        if n%2 == 1 {
            if n > odd {
                odd = n
            }
        } else if n != 0 && n < even {
            even = n
        }
    }

    if even == math.MaxInt32 {
        return odd
    }
    return odd - even
}

File: leetsync/3753-maximum-difference-between-even-and-odd-frequency-i/maximum-difference-between-even-and-odd-frequency-i.java
class Solution {
    public int maxDifference(String s) {
        int[] freq = new int[26];
        int n = s.length();
        for (int i = 0; i < n; i++) {
            char ch = s.charAt(i);
            freq[ch - 'a']++;
        }
        int oddMax = Integer.MIN_VALUE, evenMin = Integer.MAX_VALUE;
        for (int i = 0; i < 26; i++) {
            if (freq[i] == 0) {
                continue;
            }
            if (freq[i] % 2 == 0) {
                evenMin = Math.min(evenMin, freq[i]);
            } else {
                oddMax = Math.max(oddMax, freq[i]);
            }
        }
        return oddMax - evenMin;
    }
}